File: backend/src/main/java/com/sentinel/backend/entity/PermissaoPagina.java
package com.sentinel.backend.entity;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Table(name = "permissoes_pagina")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class PermissaoPagina {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String nome; // Ex: "Cadastro de Usuários"
    private String rota; // Ex: "/usuarios"
    private String metodoHttp; // Ex: GET, POST, PUT, DELETE
}

File: frontend/src/app/models/permissao-pagina.ts
export class PermissaoPagina {
  id?: number;
  nome: string;
  rota: string;
  metodoHttp: string;

  constructor(nome: string, rota: string, metodoHttp: string, id?: number) {
    this.id = id;
    this.nome = nome;
    this.rota = rota;
    this.metodoHttp = metodoHttp;
  }
}
